flink-yarn/src/test/java/org/apache/flink/yarn/YarnApplicationMasterRunnerTest.java

```diff
@@ -109,4 +120,37 @@ public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
 			taskManagerConf, workingDirectory, taskManagerMainClass, LOG);
 		assertEquals("file", ctx.getLocalResources().get("flink.jar").getResource().getScheme());
 	}
+
+	@Test
+	public void testRunAndInitializeFileSystem() throws Exception {
+		// Mock necessary system variables
+		Map<String, String> map = new HashMap<String, String>(System.getenv());
+		map.put(YarnConfigKeys.ENV_HADOOP_USER_NAME, "foo");
+		// Create dynamic properties to be used in the Flink configuration
+		map.put(YarnConfigKeys.ENV_DYNAMIC_PROPERTIES, "myKey=myValue");
+		CommonTestUtils.setEnv(map);
+
+		// Create a temporary flink-conf.yaml and to be deleted on JVM exits
+		File currDir = new File(System.getenv().get(ApplicationConstants.Environment.PWD.key()));
```

Comment:
There is already a flink-conf.yaml in the test resource directory. You can reuse it.